app/routers/api/browser.py

```
@@ -50,17 +55,50 @@ def list_directory(
+                    # For compatibility, we use relative_to with try/except
+                    resolved_path.relative_to(base)
+                    is_allowed = True
+                    break
+                except ValueError:
+                    continue
+
+            if not is_allowed:
+                safe_username = _sanitize_log_input(current_user.username)
+                safe_path = _sanitize_log_input(path)
+                logger.warning(
+                    f"Unauthorized directory browse attempt by {safe_username}: {safe_path}"
```

Ruff (0.14.14)

[warning] Logging statement uses f-string

(G004)